lib/debug/ProfilingPlugin.js

```diff
@@ -191,6 +192,24 @@ class ProfilingPlugin {
 			);
 		});
 
+		compiler.hooks.watchRun.tapAsync(pluginName, (compiler, callback) => {
+			if (this.hasRun) {
+				callback();
+				return;
+			}
+			this.hasRun = true;
+
+			// output profiling when exiting watch mode
+			process.on("SIGINT", () => {
```

Comment:
That's not something you can use here.
signals is something for the CLI not for a plugin.

Reply:
The CLI could call `close` on the Compiler, but I think it doesn't do that because that would slow down existing the process. I think you should assume that stopping the watching is undetectable. At least you are not able to do any async action on exit.

Reply:
Yes, it's very trick to detect the stop timing under current solution. I will take the idea mentioned above and add tests. 😊